src/openrct2/peep/Guest.cpp

```diff
@@ -782,14 +782,11 @@ void Guest::loc_68FA89()
                 chosen_food = bitscanforward(HasFoodExtraFlag());
                 if (chosen_food != -1)
                 {
-                    ItemExtraFlags &= ~(1 << chosen_food);
+                    RemoveItem(static_cast<ShopItem>(chosen_food));
```

Comment:
this will be removing the wrong food item. You need to add 32 to the value. In a later refactor this will not be required